SConstruct

```diff
@@ -1246,6 +1254,55 @@ else:
 env['has_demangle'] = conf.CheckDeclaration("boost::core::demangle",
                                 '#include <boost/core/demangle.hpp>', 'C++')
 
+# check BLAS/LAPACK installations
+if env["blas_lapack_libs"]:
+    for lib in env["blas_lapack_libs"]:
+        if not conf.CheckLib(lib):
+            config_error(f"Library {lib!r} could not be found.")
+
+elif env["matlab_path"] != "" and env["matlab_toolbox"] == "default":
+    # MATLAB provides the mwlapack and mwblas libraries in matlabroot/extern/lib.
+    logger.info("Skip auto-detection of Intel MKL / OpenBLAS as they are not "
+                "compatible with the MATLAB toolbox.")
+
+elif env["locate_lapack"] != "off":
+    # auto-detect versions
+    if env["locate_lapack"] in ["mkl", "auto"]:
+        blas_lapack_order = ["mkl_rt", "openblas"]
+    else:
+        blas_lapack_order = ["openblas", "mkl_rt"]
```

Comment:
The logic here isn't what I expected from the docstring for the `locate_lapack` option. I would have only expected it to try both OpenBLAS and MKL when using the `auto` option, not for that to be just the first preference with the fallback of the other. If you want to keep the behavior as it is here, then I guess the `auto` option is redundant.

Reply:
looking over this, I believe I never finished the logic here when I toggled to draft mode. After rebasing to the current `main`, I am sticking with the description of the SCons option.